exec/cpg.c

```diff
@@ -760,16 +776,25 @@ static int notify_lib_joinlist(
 		}
 	}
 
-	if (left_list_entries &&
-	    left_list[0].nodeid == api->totem_nodeid_get() &&
-	    left_list[0].reason == CONFCHG_CPG_REASON_LEAVE) {
-	qb_list_for_each(iter, &cpg_pd_list_head) {
-			struct cpg_pd *cpd = qb_list_entry (iter, struct cpg_pd, list);
-			if (mar_name_compare (&cpd->group_name, group_name) == 0 &&
-			    left_list[0].pid == cpd->pid) {
-				cpd->pid = 0;
-				memset (&cpd->group_name, 0, sizeof(cpd->group_name));
-				cpd->cpd_state = CPD_STATE_UNJOINED;
+	if (left_list_entries) {
+		/*
+		 * Zero internal cpd state for all local processes leaving group
+		 * (this cycle is not strictly needed because left_list always ether
+		 *  have exactly one process running on local node or more items
```

Comment:
s/cycle/loop
(sounds better to my non-native ears)

s/ether/either
s/have/contains (or s/have/haS)

Reply:
Czechlish. Fixed